### PR TITLE
2024 05 datamodel

### PR DESCRIPTION
**Pull request by sjib**
- [x] Datamodel with selected classes from SIA405 Fernwärme 03_tdh_db_sia405.sql
- [x] Datamodel dictionaries 09_tdh_dictionaries.sql
- [x] Datamodel SIA405 Fernwirkkabel 03_tdh_control_cable_db_sia405.sql
- [x] Datamodel SIA405 Fernwirkkabel Dictionaries
- [x] Datamodel class organisation 03_tdh_db_organisation.sql
- [x] Datamodel class organisation dictionaries 09_tdh_organisation_dictionaries.sql
- [x] Relations to organisation (needed to set fk_owner, additional fk_* see https://github.com/teksi/district_heating/issues/5
- [x] setup scripts
- [x] oid_generation
- [x] default values fk_dataowner, fk_provider, fk_owner
- [ ] CI infrastructure
- [ ] check 12_0_roles.sql / 12_1_roles.sql  - latest adaptions from wastewater

### CODE REVIEW

**Comment by sjib** [comment]
@ponceta How can I activate the workflows?

**Comment by ponceta** [comment]
AS you see here :
https://github.com/teksi/distance_heating/pull/1/files#diff-fc5dead48e4c3eacc43a4bbdf67b2c57eb59b976d541c35a6c102c43a3a133ecR7-R18

Workflows will trigger on certain parameters.

* distance_heating had the old naming for the "main" branch which was called "master"

I've fixed that and now you should be able to trigger and fix actions.

I'm available if you need me today! ;)

**Comment by sjib** [comment]
`Run ./datamodel/test/static_tests.sh
  ./datamodel/test/static_tests.sh
  shell: /usr/bin/bash -e {0}
/home/runner/work/_temp/7a9068bb-6c98-45bb-bde4-12cc02532249.sh: line 1: ./datamodel/test/static_tests.sh: Permission denied
Error: Process completed with exit code 126.`

@ponceta Can you check the permissions?

**Comment by sjib** [comment]
Salut Stefan,

I was able to download the project files for github.

Please note that we work in a windows environment and I had to copy the setup.sh (for linux environment)  to setup.cmd.
I do not know if this environment must be taken into account for TEKSI users according to the product strategy. 
If not, this would be an advantage.

**Comment by sjib** [comment]
In the 03_tdh_db_sia405.sql, I had comment out int the relationship section  the following lines since the refrences do not exist :

```
--ALTER TABLE tdh_od.pipe_section ADD CONSTRAINT rel_pipe_section_hydraulic_line_section FOREIGN KEY (fk_hydraulic_line_section) REFERENCES tdh_od.hydraulic_line_section(obj_id) ON UPDATE CASCADE ON DELETE set null DEFERRABLE INITIALLY DEFERRED; 
--ALTER TABLE tdh_od.pipe_section ADD CONSTRAINT rel_pipe_section_static_line_section FOREIGN KEY (fk_static_line_section) REFERENCES tdh_od.static_line_section(obj_id) ON UPDATE CASCADE ON DELETE set null DEFERRABLE INITIALLY DEFERRED; 
--ALTER TABLE tdh_od.pipe_point ADD CONSTRAINT rel_pipe_point_hydraulic_node FOREIGN KEY (fk_hydraulic_node) REFERENCES tdh_od.hydraulic_node(obj_id) ON UPDATE CASCADE ON DELETE set null DEFERRABLE INITIALLY DEFERRED; 
--ALTER TABLE tdh_od.pipe_point ADD CONSTRAINT rel_pipe_point_static_node FOREIGN KEY (fk_static_node) REFERENCES tdh_od.static_node(obj_id) ON UPDATE CASCADE ON DELETE set null DEFERRABLE INITIALLY DEFERRED;
```

**Comment by ponceta** [comment]
There's something not set for docker to run properly. I need @3nids or @domi4484 assistance to solve this.

**Comment by ponceta** [comment]
`OCI runtime exec failed: exec failed: unable to start container process: exec: "init_db.sh": executable file not found in $PATH: unknown`

**Comment by sjib** [comment]
@jpdupuy Can you test this new version of the data model and dictionaries.

I still have to suppress the attributes owner (too much) as there are now already fk_owner and relations.

File: datamodel/changelogs/0001/03_tdh_db_sia405.sql [hunk 1]
```diff
@@ -0,0 +1,861 @@
...
+WITH (
+   OIDS = False
+);
+CREATE SEQUENCE tdh_od.seq_pipe_point_oid INCREMENT 1 MINVALUE 0 MAXVALUE 999999 START 0;
+ ALTER TABLE tdh_od.pipe_point ALTER COLUMN obj_id SET DEFAULT tdh_sys.generate_oid('tdh_od','pipe_point');
+COMMENT ON COLUMN tdh_od.pipe_point.obj_id IS 'INTERLIS STANDARD OID (with Postfix/Präfix), see www.interlis.ch';
+--ALTER TABLE tdh_od.pipe_point ADD COLUMN geometry_geometry geometry('POINT', :SRID);
+-- CREATE INDEX in_tdh_pipe_point_geometry_geometry ON tdh_od.pipe_point USING gist (geometry_geometry );
+-- COMMENT ON COLUMN tdh_od.pipe_point.geometry_geometry IS '';
+ALTER TABLE tdh_od.pipe_point ADD COLUMN geometry3d_geometry geometry('COMPOUNDCURVEZ', :SRID);
```
Comment (ponceta) [inline datamodel/changelogs/0001/03_tdh_db_sia405.sql:117]:
```suggestion
ALTER TABLE tdh_od.pipe_point ADD COLUMN geometry3d_geometry geometry('COMPOUNDCURVEZ', :SRID);
```

**Comment by sjib** [comment]
@jpdupuy Merci for the adaptions in organisation dictionaries

File: datamodel/changelogs/0001/03_tdh_db_sia405.sql (hunk 1 above)
Comment (sjib) [reply datamodel/changelogs/0001/03_tdh_db_sia405.sql]: Change to POINTZ

**Pull request merged**